components/post-title/index.tsx

```diff
@@ -37,10 +46,6 @@ export const PostTitle = (props) => {
 	);
 };
 
-PostTitle.propTypes = {
-	tagName: PropTypes.string,
-};
-
 PostTitle.defaultProps = {
 	tagName: 'h1',
```

Comment:
Lets move this to the inline assignment (default value) so we can remove prop types completely

Reply:
@fabiankaegy this PR isn't ready yet, I'm having a hard time assigning that as a default value.

I get this error:

```
Type '"h1"' is not assignable to type 'TElementType'.
  '"h1"' is assignable to the constraint of type 'TElementType', but 'TElementType' could be instantiated with a different subtype of constraint 'ElementType<any>'.ts(2322)
```

which makes sense but unsure on how to resolve it. I'll mark this PR as draft.